otwcode/otwarchive
# AO3-5355 Add filtering bookmarks by bookmarker's tag names

## Issue

https://otwarchive.atlassian.net/browse/AO3-5355

## Purpose

These new fields now work:

- Other bookmarker's tags to include
- Other bookmarker's tags to exclude

Previously we have these fields on the BookmarkSearchForm options:

- tag_ids: IDs of bookmarker tags, handled by BookmarkQuery, search on bookmarks
- other_tag_names: handled by TaggableQuery, search on bookmarkables, but not bookmarks (causing [AO3-5317](https://otwarchive.atlassian.net/browse/AO3-5317))
- excluded_tag_ids/excluded_tag_names: handled by both BookmarkQuery and BookmarkableQuery through TaggableQuery, search on bookmarkables and bookmarks

I refactored so the new fields are as follows:

- <tag_type>_ids/other_tag_names: for bookmarkables (similar to work filters, this is the shared, unchanged logic of TaggableQuery)
- excluded_tag_ids/excluded_tag_names: for bookmarkables (same fields, still handled by TaggableQuery)
- tag_ids/other_bookmark_tag_names: for bookmarks; the former is already there, the latter is new
- excluded_bookmark_tag_ids/excluded_bookmark_tag_names: for bookmarks; the former is now separated from excluded_tag_ids, the latter is new

Now, **only** the "Bookmarker's Tags" checkboxes and the 2 "Other bookmarker's tags to include/exclude" fields will be used to search tags on bookmarks, everything else will apply to bookmarkables.

All bookmark-specific fields will be processed in BookmarkQuery, the rest in BookmarkableQuery/TaggableQuery. BookmarkQuery no longer needs to include TaggableQuery.

There's another new field, "Search within bookmarker's tags and notes", which I'm leaving out of this pull request.

## Testing

- Try filtering using all 4 "other" tag fields in the bookmark filters
- Try filtering using the "Bookmarker's Tags" checkboxes, include/exclude
- See [AO3-5317](https://otwarchive.atlassian.net/browse/AO3-5317)
